solr/core/src/java/org/apache/solr/update/processor/DistributedUpdateProcessor.java

```diff
@@ -1032,6 +1151,24 @@ protected void assertNotFinished() {
     finished = true;
   }
 
+  /**
+   *
+   * @return whether this update changes a value of a block
+   */
+  public static boolean shouldRefreshUlogCaches(AddUpdateCommand cmd) {
```

Comment:
Can we make this logic more efficient by adding a field to AddUpdateCommand that declares (a) is this update an atomic update, and furthermore, did it involve nested documents?  Perhaps two separate `Boolean`s, null meaning not yet calculated.  By this juncture in the code I think we've already determined these things and I'd like to avoid potential overhead per document going into Solr.  Lambda's can add up.

typo "update.q" => "update"